solr/core/src/java/org/apache/solr/handler/component/HttpShardHandlerFactory.java

```diff
@@ -149,7 +152,9 @@
    */
   @Override
   public ShardHandler getShardHandler() {
-    return getShardHandler(defaultClient);
+    return getShardHandler(defaultClient != null?
+            defaultClient:
+            multipleClients.get( r.nextInt(multipleClients.size()) % multipleClients.size()));
```

Comment:
minor nit, we probably don't need `% multipleClients.size()` as r.nextInt(n) returns 0 (inclusive) to n - 1

Reply:
true.